inputgraph/loader.go

```diff
@@ -22,7 +23,10 @@ import (
 	"github.com/pkg/errors"
 )
 
-var ErrRemoteNotSupported = fmt.Errorf("remote targets not supported")
+var (
+	errCannotLoadRemoteTarget  = errors.New("remote targets not supported")
+	errUnsupportedRemoteTarget = errors.New("only remote targets with full SHAs or tags are supported")
```

Comment:
I would clarify a little better that only tags explicitly referenced with the `tags/` prefix are supported.

```suggestion
	errUnsupportedRemoteTarget = errors.New("only remote targets with full SHAs or tags referenced explicitly as 'tags/<my-tag>' are supported")
```

Reply:
maybe we can start using hinterror type from now on instead of having to go back later and make these clearer.
This way the user might see in the end:
```
Help: The remote target url is <url>, Please make sure your remote target is referenced in the form of `url/repo:full_sha` or  `url/repo/tags/<my-tag>`
```

Reply:
Yeah, that would be nice!

Reply:
Ok, sounds good.

Reply:
It's not obvious to me where this text belongs. Inline, before the log sharing link, at the very end?
![image](https://github.com/earthly/earthly/assets/332408/fc425455-5cad-4f4c-a45d-f6d7d2c560ee)

Reply:
oh, honestly I forgot that this is not a build error but more of a warning for why autoskip would not work, so not sure how this is going to be displayed.

Reply:
Since auto-skip is now opt-in, as opposed to default on for pipelines, I wonder if the auto-skip errors should just be fatal, and not just warnings.

I would imagine that most people who use auto-skip will want to make sure that the build's compatibility with auto-skip is enforced.

Maybe we can still offer a flag to ignore auto-skip errors, if someone still wants the previous behavior.

Reply:
Sure, that seems reasonable.

Reply:
On second thought, that change will require some test & output changes. Can we merge this PR first? I clarified the error message and added a tag example.